common/src/main/java/gov/cms/ab2d/common/service/CoverageService.java

```diff
@@ -19,11 +20,20 @@
 
     /**
      * Get {@link CoveragePeriod} in database matching provided triple
-     * @param contractId existing {@link gov.cms.ab2d.common.model.Contract#getId()}
+     * @param contract existing {@link gov.cms.ab2d.common.model.Contract#getId()}
      * @param month valid month
      * @param year valid year (not later than current year)
      */
-    CoveragePeriod getCoveragePeriod(long contractId, int month, int year);
+    CoveragePeriod getCoveragePeriod(Contract contract, int month, int year);
+
+    /**
+     * Create {@link CoveragePeriod} matching the provided triple
+     * @param contract contract to add coverage period for
+     * @param month month of the period
+     * @param year year of the period
+     * @return coverage period as it exists in database
+     */
+    CoveragePeriod getOrCreateCoveragePeriod(Contract contract, int month, int year);
```

Comment:
While descriptive, quite a clunky name.

From https://stackoverflow.com/questions/3184197/whats-a-good-name-for-a-method-that-gets-or-creates-an-object, some ideas:

getsertCoveragePeriod
acquireCoveragePeriod
obtainCoveragePeriod